bin/pip_find_builddeps.py

```diff
@@ -56,47 +55,48 @@ def _filter_builddeps(pip_download_output_file):
     return sorted(builddeps)
 
 
-@contextlib.contextmanager
-def _remove_on_success(tmpdir):
-    """Remove tmpdir only if no exception is raised during context."""
-    try:
-        yield
-    except:  # noqa: E722
-        # Re-raise the exception but keep the tmpdir
-        raise
-    else:
-        shutil.rmtree(tmpdir)
-
-
-def find_builddeps(requirements_files, no_cache=False):
+def find_builddeps(requirements_files, no_cache=False, ignore_errors=False):
     """
     Find build dependencies for packages in requirements files.
 
     :param requirements_files: list of requirements file paths
     :param no_cache: do not use pip cache when downloading packages
-    :return: list of build dependencies (in requirements.txt format)
+    :param ignore_errors: generate partial output even if pip download fails
+    :return: list of build dependencies and bool whether output is partial
     """
     tmpdir = tempfile.mkdtemp(prefix=f"{SCRIPT_NAME}-")
+    pip_output_file = Path(tmpdir) / "pip-download-output.txt"
+    is_partial = False
 
-    with _remove_on_success(tmpdir):
-        pip_output_file = Path(tmpdir) / "pip-download-output.txt"
-
-        try:
-            log.info("Running pip download, this may take a while")
-            _pip_download(requirements_files, pip_output_file, tmpdir, no_cache)
-        except subprocess.CalledProcessError:
-            msg = f"pip download failed, see {pip_output_file} for more info"
+    try:
+        log.info("Running pip download, this may take a while")
+        _pip_download(requirements_files, pip_output_file, tmpdir, no_cache)
+    except subprocess.CalledProcessError:
+        msg = f"Pip download failed, see {pip_output_file} for more info"
+        if ignore_errors:
+            log.error(msg)
+            log.warning("Ignoring error...")
+            is_partial = True
+        else:
             raise FindBuilddepsError(msg)
 
+    try:
         log.info("Looking for build dependencies in the output of pip download")
-        return _filter_builddeps(pip_output_file)
+        builddeps = _filter_builddeps(pip_output_file)
```

Comment:
Not sure if I follow the reasoning behind this last try/finally block: Let's say an encoding error happened while parsing the `pip download` output even if the command succeeded. Then we are still removing the file before raising the exception, right? Is this the intended behaviour?

Reply:
Hmm, true, if parsing the file fails, it would probably be nice to keep that file. Will change